R/lint.R

```diff
@@ -259,6 +259,8 @@ lint_dir <- function(path = ".", relative_path = TRUE, ..., exclusions = list("r
 #' Apply one or more linters to all of the R files in a package.
 #' @param path the path to the base directory of the package, if \code{NULL},
 #' it will be searched in the parent directories of the current directory.
+#' @param path_prefix a prefix to add to the path of any file found, useful for
```

Comment:
I see the idea of what you're after & it makes sense, I'm less confident this is the best API for it. Maybe use `project_dir` instead, and then `path` is given relative to that?

cc @russHyde and @AshesITR for an API decision

Reply:
Thanks for the feedback, I would be happy to do that if that's the direction y'all would like.

For what it's worth, when we run these in our CI, we actually run them from the subdirectory, so having `path_prefix` / `project_dir` only add on to the output of the GHA annotations is the behavior we want (of course we could change where we lint from easily, so that's not a great argument for this API). If we did keep this behavior, we might also consider naming it something like `annotation_path_prefix` to make it's clear it's only about annotations and not where to run the file.

Reply:
IMO this is not really related to linting per se, but more to the `print.lints` method for GitHub.